### PR TITLE
[JENKINS-63580] Update regex for msbuild to recognize linker warnings

### PR DESCRIPTION
**Pull request by pyatizbyantsevia**
Closes JENKINS-63580

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: SUPPORTED-FORMATS.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<!--- DO NOT EDIT -- Generated at 2023-09-20T10:55:02.107301 - Run the `main` method of `ParserRegistry` to regenerate after changing parsers -- DO NOT EDIT --->
+<!--- DO NOT EDIT -- Generated at 2023-10-03T10:06:45.445947 - Run the `main` method of `ParserRegistry` to regenerate after changing parsers -- DO NOT EDIT --->
```
Comment (uhafner) [inline SUPPORTED-FORMATS.md:1]: Can you please remove that file from the change list?

**Review by uhafner — APPROVED** [review]
It seems that the regular expression actually is degenerated, too many options have been added year by year. I tried to fix that a little bit, and now your test works :-) 

(Maybe it makes sense to start from scratch sometimes... But for now the PR is ready to be integrated, thanks!)

**Comment by pyatizbyantsevia** [comment]
Thank you very much for making changes to my test, I focused on the parsing problem (on the first assert) and did not bring the test itself to the proper form.
P.s. Before accepting PR, I kindly ask you to put a label for Hacktoberfest =)

**Review by uhafner — APPROVED** [review]

File: src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java [hunk 2]
```diff
@@ -22,13 +22,22 @@ public class MsBuildParser extends LookaheadParser {
...
-            + "\\s*:\\s(?:\\s*([A-Za-z0-9.]+)\\s*:)?\\s*(.*?)(?: \\[([^\\]]*)[/\\\\][^\\]\\\\]+\\])?"
-            + "|(.*)\\s*:.*error\\s*(LNK[0-9]+):\\s*(.*)))$";
-
-    private final Pattern ignoredToolsPattern = Pattern.compile("(?!.exe)(\\.[^.]+)$");
+            = "(?:^(?:.*)Command line warning ([A-Za-z0-9]+):\\s*(.*)\\s*\\[(.*)\\])"
+            + "|"
+            + ANT_TASK + "(?:"
+            + "(?:\\s*(?:\\d+|\\d+:\\d+)>)?(?:(?:(?:(.*?)\\((\\d*)(?:,(\\d+))?[a-zA-Z]*?\\)|.*LINK)"
+            + "\\s*:|"
+            + "(.*):)\\s*([A-z-_]*\\s(?:[Nn]ote|[Ii]nfo|[Ww]arning|(?:fatal\\s*)?[Ee]rror))[^A-Za-z0-9]\\s*"
```
[the remark this answers is not in the document]
Comment (pyatizbyantsevia) [reply src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java:30]: (Sorry, my comment was marked as PENDING and I didn't realize it wasn't visible)

[the remark this answers is not in the document]
Comment (uhafner) [reply src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java:30]: While I stepped into the code during debugging I noticed that the last part of the regexp wasn't used at all. So I deleted that part totally. (Actually I am not the author of the regexp and only tested it a little bit in the debugger. In the long term it should be simplified).

[the remark this answers is not in the document]
Comment (pyatizbyantsevia) [reply src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java:30]: Thanks for the answer! I get it. I haven’t learned how to debug plugins yet, so I only tested everything by running tests =)

**Comment by uhafner** [comment]
Ok, thanks, released in https://github.com/jenkinsci/analysis-model/releases/tag/v11.9.0

**Pull request merged**